drivers/unix/socket_helpers.h

```diff
@@ -3,6 +3,13 @@
 
 #include <string.h>
 
+#if defined(__MINGW32__ ) && (!defined(__MINGW64_VERSION_MAJOR) || __MINGW64_VERSION_MAJOR < 4)
+  // Workaround for mingw-w64 < 4.0
```

Comment:
Actually it's only for mingw-w64 < 3.0 ;) (or at least in the 3.x branch on their git repo the define is there).

Could you squash it all in the first commit, to avoid removing and readding the workaround in two commits?